src/psyclone/psyir/symbols/containersymbol.py

```diff
@@ -213,43 +212,36 @@ class FortranModuleInterface(ContainerSymbolInterface):
 
     @staticmethod
     def import_container(name):
-        ''' Imports a Fortran module as a PSyIR container. The module is
-        expected to be found in a Fortran source file with the same name
-        as the module plus the '.[f|F]90' extension. The search
-        locations are provided in-order by the Config include_paths
-        attribute ('-I' in the psyclone script).
+        ''' Imports a Fortran module as a PSyIR container via the
+        ModuleManager.
 
         :param str name: name of the module to be imported.
 
         :returns: container associated with the given name.
         :rtype: :py:class:`psyclone.psyir.nodes.Container`
 
-        :raises SymbolError: the given Fortran module is not found on the \
+        :raises SymbolError: the given Fortran module is not found on the
             import path.
 
         '''
-        # pylint: disable=import-outside-toplevel
-        from psyclone.psyir.frontend.fortran import FortranReader
-        for directory in Config.get().include_paths:
-            for filename in [name+'.f90', name+'.F90']:
-                if filename in listdir(directory):
-                    # Parse the module source code
-                    abspath = path.join(directory, filename)
-                    fortran_reader = FortranReader()
-                    file_container = fortran_reader.psyir_from_file(abspath)
-                    # Check the expected container is in this file
-                    for candidate in file_container.children:
-                        if candidate.name.lower() == name.lower():
-                            return candidate
-                    raise ValueError(
-                        f"Error importing the Fortran module '{name}' into a "
-                        f"PSyIR container. The file with filename "
-                        f"'{filename}' does not contain the expected module.")
-
-        raise SymbolError(
-            f"Module '{name}' (expected to be found in '{name}.[f|F]90') not "
-            f"found in any of the include_paths directories "
-            f"{Config.get().include_paths}.")
+        # pylint: disable-next=import-outside-toplevel
+        from psyclone.parse import ModuleManager
+        mod_manager = ModuleManager.get()
+
+        # TODO #2011 - rationalise how this interacts with the kernel search
+        # path set in generate().
+        mod_manager.add_search_path(Config.get().include_paths)
```

Comment:
Great change, I love the use of the ModuleManager (of course I am biased :) )

Reply:
I'll take praise where I can get it :-)